crates/yamlpath/src/lib.rs

```diff
@@ -272,6 +272,7 @@ enum QueryMode {
 }
 
 /// Represents a queryable YAML document.
+#[derive(Clone)]
```

Comment:
NB: I don't love that this is now `Clone`; the only reason I did this is to make the "loop while creating new documents" pattern a bit simpler. But I'm going to try and remove it.